### DISCUSSION TITLE
AtomRebirth experiments - what's being done, and what's all about?

### DISCUSSION THREAD

**Discussion by mauricioszabo** [atom-ide-community/atom#400] [Show and tell]
So, I created the https://github.com/AtomRebirth/atom repo just when I heard about the sunset. It's for stuff varying from experimental to HIGHLY experimental (almost what people consider "nightly" builds).

The idea is to have all of the code there being upstreamed to this repo. In the meantime, I set up a CI to build on each push, so I can provide binaries for people to test (myself included) this versions. As of now, I have a version that runs Electron 12 and current tree-sitter (https://cirrus-ci.com/build/6443769909215232), without the Crash Report (need work to change) and the github package (broke for some reason - need to fix it too). Also, it can't detect _core themes_ so some if you navigate to packages you'll see no theme bundled with Atom (working on it too). But with the version-bump, Apple M1 works (there's a binary there).

I **do not want** to keep a separate fork of Atom - this repo is to test ideas. After an idea is tested and hopefully validated, it'll be upstreamed back to `atom-community` repo (this one) and pass all linter, quality, unit tests and other requirenments

Also, there will be a way to install both versions - the atom-community and the AtomRebirth - on the same machine. In that way, people (me included) can test if performance got better, got worse, if there's something breaking in only one of them, etc. I'll keep posting things here with my experiments, if someone wants to jump in :). 

Also, if someone is worried about installing Atom from an "unknown source", I can understand. If there's something I can do to add some kind of security so people know that my code is not trying to do shady things, please ping me and I'll try to implement this too :).

**Comment by ThatXliner** [comment]
Cool! two questions:

 - What's blockmap (in the cirrus CI binaries. at least for macos)
 - Any chance of setting up the CI here as well?

**Reply by meadowsys** [reply]
here we have Azure pipelines for CI already, so I'm not quite sure if there would be a need / want to move stuff over? ~~though I would like to migrate off of Microsoft infrastructure personally~~

**Reply by ThatXliner** [reply]
Sorry, I meant to say CI for binaries.

**Reply by DeeDeeG** [reply]
Azure Pipelines makes it a little hard to find the built Atom packages/binaries, but they are there...

Steps: Visit [here](https://dev.azure.com/atomcommunity/atomcommunity/_build/latest?definitionId=10&branchName=master) --> Click the "**8 published**" link/button\* --> That takes you to [the downloads](https://dev.azure.com/atomcommunity/atomcommunity/_build/results?buildId=1183&view=artifacts&pathAsName=false&type=publishedArtifacts).

_\* Note: You can do Find in Page (Ctrl + F), and search for "published", to make the published binaries link a bit easier to find._

**Reply by ThatXliner** [reply]
Oh, so we *have* binaries?

**Reply by DeeDeeG** [reply]
His fork does build more targets, though. Such as AppImage on Linux, and arm64 macOS builds.

**Reply by mauricioszabo** [reply]
Answering the questions: I'm not sure what is the blockmap, honestly- it's something that electron-builder produced, and the CI just copied to the artifacts

As for publishing the packages here, it's possible but to be honest, these are not intended to be "stable" releases. They can (and probably WILL) diverge from the master branch, so you can have a binary that have recent Electron, one that have the old one, a binary with all packages, other with none, mainly because I'm testing things. This means that it can be confusing to know what a binary includes, as with the CI these are easier because they are grouped by branch name...

**Reply by mauricioszabo** [reply]
About the new builds: they actually are not working correctly - for some reason, some keybindings don't work and there are no syntaxes available to change. I'll look at this issue too...

**Comment by lifeisbard** [comment]
I tried to run the macOS ARM binary, but Monterey alerted me that it is broken and advised that I move it to the trash. Did see the "Apple" indicator instead of "Intel" in "Activity Monitor", which gives me some hope that maybe Atom Community will run on M1 soon.

**Reply by mauricioszabo** [reply]
There is an experimental build on my fork of Atom: https://cirrus-ci.com/task/5997043503071232

It's not 100% stable yet, but if you want to take a look to see if it works for you, look for `binary` on the Artifacts session

**Reply by lifeisbard** [reply]
Hi, thanks for your reply, and thanks for putting this together. Have you changed anything on it in the past three days? I tried a few days ago but macOS Monterey has rejected it. I commented above regarding the matter.

Thanks 🙏🏻
